tools/discord/bot.py

```diff
@@ -0,0 +1,246 @@
+import asyncio
+import discord
+from dotenv import load_dotenv
+import os
+from typing import Optional
+
+# Global variable management
+load_dotenv()
+TOKEN = os.getenv("token")
+DOLPHIN_PATH = os.getenv("dolphin_path")
+SP_PATH = os.getenv("sp_path")
+NAND_PATH = os.getenv("nand_path")
+IS_GENERATING_KRKG = False
+
+client = discord.Client(intents=discord.Intents.default())
+tree = discord.app_commands.CommandTree(client)
+
+# ================================
+#     DOLPHIN
+# ================================
+
+
+async def dolphin_clear_io():
+    global NAND_PATH
+
+    rkg_path = os.path.join(NAND_PATH, "test.rkg")
+    krkg_path = os.path.join(NAND_PATH, "test.krkg")
+    ok_path = os.path.join(NAND_PATH, "ok")
+    fail_path = os.path.join(NAND_PATH, "fail")
+
+    if os.path.exists(rkg_path):
+        os.remove(rkg_path)
+
+    if os.path.exists(krkg_path):
+        os.remove(krkg_path)
+
+    if os.path.exists(ok_path):
+        os.remove(ok_path)
+
+    if os.path.exists(fail_path):
+        os.remove(fail_path)
+
+
+async def dolphin_set_ghost(ghost: bytes):
+    global NAND_PATH
+
+    with open(os.path.join(NAND_PATH, "test.rkg"), "wb") as f:
+        f.write(ghost)
+
+
+async def dolphin_run():
+    global DOLPHIN_PATH, SP_PATH
+
+    proc = await asyncio.create_subprocess_exec(DOLPHIN_PATH, "-e", SP_PATH, "-b")
```

Comment:
Remove frame limit. Just in case a new Dolphin version or new env and you forget to change!
```suggestion
    proc = await asyncio.create_subprocess_exec(DOLPHIN_PATH, "-e", SP_PATH, "-b", "-C", "Dolphin.Core.EmulationSpeed=0")
```

Reply:
Can add `-C Dolphin.Display.RenderToMain=true` for silent Dolphin run (no window is created)

Reply:
Add `-C Dolphin.DSP.Backend="No Audio Output"` for no audio output

Reply:
Added.